validphys2/src/validphys/core.py

```diff
@@ -266,83 +264,153 @@ def get_kinlabel_key(process_label):
                          "variables matching  the process %s Check the "
                          "labels defined in commondata.cc. " % (l)) from e
 
-CommonDataMetadata = namedtuple('CommonDataMetadata', ('name', 'nsys', 'ndata', 'process_type'))
 
-def peek_commondata_metadata(commondatafilename):
-    """Check some basic properties commondata object without going though the
-    trouble of processing it on the C++ side"""
-    with open(commondatafilename) as f:
-        try:
-            l = f.readline()
-            name, nsys_str, ndata_str = l.split()
-            l = f.readline()
-            process_type_str = l.split()[1]
-        except Exception:
-            log.error(f"Error processing {commondatafilename}")
-            raise
+class CommonDataSpec(TupleComp):
 
-    return CommonDataMetadata(name, int(nsys_str), int(ndata_str),
-                              get_kinlabel_key(process_type_str))
+    def __init__(self, name, variant, metadata_file=None):
+        parent = metadata_file.parent
+        metadata = yaml.safe_load(metadata_file.open("r", encoding="utf-8"))
+
+        # Prepare to read the defaults
+        dat = metadata["data_central"]
+        unc = metadata["data_uncertainties"]
+        kin = metadata["data_kinematics"]
+
+        # Check whether the variants override any of the files
+        # TODO: various try-except in case things don't work as expected
+        if variant is None:
+            variant_info = metadata[variant]
```

Comment:
Is `variant` a required key then?

Reply:
I guess the question/decision is "it's default a variant?". I like to think it is but I don't have a strong opinion. I should prepare a list with these points.